modules/tensor_mechanics/include/materials/abaqus/AbaqusUMATStress.h

```diff
@@ -10,17 +10,61 @@
 #pragma once
 
 #include "ComputeStressBase.h"
+#include "ComputeLagrangianObjectiveStress.h"
 #include "DynamicLibraryLoader.h"
 
+// forward declaration
+template <bool>
+class AbaqusUMATStressBase;
+
+// Shim for the old system
+template <>
+class AbaqusUMATStressBase</*new_system = */ false> : public ComputeStressBase
+{
+public:
+  static InputParameters validParams() { return ComputeStressBase::validParams(); }
+
+  AbaqusUMATStressBase(const InputParameters & params)
+    : ComputeStressBase(params),
+      _stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "stress"))
+  {
+  }
+
+protected:
+  virtual void computeQpStress() { computeQpStressHelper(); }
+  virtual void computeQpStressHelper() = 0;
+  const MaterialProperty<RankTwoTensor> & _stress_old;
+};
+
+// Shim for the new system
+template <>
+class AbaqusUMATStressBase</*new_system = */ true> : public ComputeLagrangianObjectiveStress
+{
+public:
+  static InputParameters validParams() { return ComputeLagrangianObjectiveStress::validParams(); }
+
+  AbaqusUMATStressBase(const InputParameters & params)
+    : ComputeLagrangianObjectiveStress(params),
```

Comment:
I think this could be made a lot more compact with 

```
template <bool new_system>
class AbaqusUMATStressTempl : public template std::conditional<new_system, ComputeLagrangianObjectiveStress, ComputeStressBase>::type
```

and

```
_stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + (new_system ? "cauchy_stress" : "stress")))
```

uhm, if it wasn't for `computeQpStress` vs. `computeQpSmallStress`... :-(

Reply:
Yep, this is exactly what I started with, until I realized I need two different methods.

Reply:
Can we rename all `computeQpSmallStress` to `computeQpStress`?